sherpa/plot/__init__.py

```diff
@@ -75,24 +77,44 @@
 plot_opt = config.get('options', 'plot_pkg', fallback='dummy')
 plot_opt = [o.strip().lower() for o in plot_opt.split()]
 
-plot_backends = {'pylab': 'sherpa.plot.pylab_backend.PylabBackend',
-                 'matplotlib': 'sherpa.plot.pylab_backend.PylabBackend',
-                 'dummy': 'sherpa.plot.backends.BasicBackend',
+plot_backends = {'pylab': ['sherpa.plot.pylab_backend', 'PylabBackend'],
+                 'matplotlib': ['sherpa.plot.pylab_backend', 'PylabBackend'],
+                 'dummy': ['sherpa.plot.backends', 'BasicBackend'],
                  }
 
 backend = None
 '''Currently active backend module for plotting.'''
 
+basicbackend = BasicBackend()
+'''This backend has all the defaults that are backend-independent.
+For all the plot classes in this module (e.g. FitPlot, JointPlot, ...)
+the default settings are set in code like `plot_prefs = xxx.get_plot_defaults()`
+where xxx is some backend. In the current design, this code is executed when this
+module imported and then the values are shared between all instances of a class.
+That allows one to change those default "globally" (i.e. for all instances of the class)
+but it means that the default values are based on the backend that is active when
+this module is imported. That depends on the sherpa.rc setting and thus `plot_prefs`
+might contain e.g. matplotlib specific defaults that are not applicable when other 
+backends are used.
+Thus, we currently initialize them with the BasicBackend that has only the
+backend-indenpendent (i.e. those that works for any backend) defaults set.
+
+Note that this kind of indicates that the defaults don't have to be taken from a
+backend - they are statics to the class and could live just there, but that is 
+a design change that needs to be discussed in more detail.
+'''
+
 for plottry in plot_opt:
     try:
-        backend_class = importlib.import_module()
+        backend_module = importlib.import_module(plot_backends[plottry][0])
+        backend = getattr(backend_module, plot_backends[plottry][1])()
         break
     except ImportError:
         pass
 else:
     # None of the options in the rc file work, e.g. because it's an old file
     # that does not have dummy listed
-    import sherpa.plot.dummy_backend as backend
+    backend = BasicBackend()
```

Comment:
Should this also set up `backend_module`?

Reply:
What is `backend_module` meant to be?

Reply:
I assume it was a symbol present in at least one version of the code. Or I got very confused. I think I was assuming that the backend names were still being used in the file names, but this is not the case (and we may want to allow multiple backends in a module, or have aliases, in case we want to rename `pylab` to `matplotlib` or some other such name in the future.

Anyway, I think this can be closed now, but I'll leave it open for a bit.